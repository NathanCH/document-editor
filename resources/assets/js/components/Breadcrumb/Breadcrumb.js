import React from 'react';
import PropTypes from 'prop-types';

import './Breadcrumb.scss';

const Breadcrumb = (prop) => (
  <ul className="breadcrumb">
    <li className="breadcrumb-item">
      <a href="/" className="breadcrumb-link">
        <i className="fa fa-home"></i>
      </a>
    </li>
    {prop.current &&
      <li className="breadcrumb-item active">
        {prop.current}
      </li>
    }
  </ul>
);

Breadcrumb.propTypes = {
  current: PropTypes.string,
};

Breadcrumb.defaultProps = {
  current: null,
};

export default Breadcrumb;
