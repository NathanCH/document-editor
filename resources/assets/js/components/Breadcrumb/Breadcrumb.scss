.breadcrumb {
  background: none;
  padding: 0.75rem 0;
}