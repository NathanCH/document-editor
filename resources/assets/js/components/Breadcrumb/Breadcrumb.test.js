import React from 'react';
import renderer from 'react-test-renderer';

import Breadcrumb from './Breadcrumb';

describe('<BreadCrumb />', () => {
  test('it renders without props', () => {
    const component = renderer
      .create(<Breadcrumb />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('it renders `props.current`', () => {
    const component = renderer
      .create(<Breadcrumb current="Test" />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('it renders `active` class on current breadcrumb ', () => {
    const component = renderer.create(
      <Breadcrumb current="Test" />
    );

    let instance = component.root;

    const item = instance.find(
      el => el.type == 'li'
        && el.props.className.includes('active')
    );

    expect(item).toBeInstanceOf(Object);
  });

});