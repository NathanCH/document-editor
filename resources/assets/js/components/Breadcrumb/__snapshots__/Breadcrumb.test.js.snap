// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadCrumb /> it renders \`props.current\` 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb-item"
  >
    <a
      className="breadcrumb-link"
      href="/"
    >
      <i
        className="fa fa-home"
      />
    </a>
  </li>
  <li
    className="breadcrumb-item active"
  >
    Test
  </li>
</ul>
`;

exports[`<BreadCrumb /> it renders without props 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb-item"
  >
    <a
      className="breadcrumb-link"
      href="/"
    >
      <i
        className="fa fa-home"
      />
    </a>
  </li>
</ul>
`;
